python/lsst/ts/eas/eas_csc.py

```diff
@@ -154,6 +159,34 @@ async def handle_summary_state(self) -> None:
 
             await self.disconnect()
 
+    async def end_disable(self, id_data: salobj.BaseDdsDataType) -> None:
+        """End do_disable; called after state changes but before command
+        acknowledged.
+
+        This method acknowledges the do_disable command and stops the
+        M1M3TS control loop.
+
+        Parameters
+        ----------
+        id_data: `CommandIdData`
+            Command ID and data
+        """
+
+        await self.cmd_disable.ack_in_progress(id_data, timeout=SAL_TIMEOUT)
+        try:
+            async with self.control_loop_lock:
+                await asyncio.wait(
```

Comment:
This would be a lot cleaner if you used `asyncio.wait_for` instead.

```
await asyncio.wait_for(self.m1m3_stop(), timeout=M1M3TS_STOP_TIMEOUT)
```